packages/pangea-sdk/pangea_django/pangea_django_auth.py

```diff
@@ -0,0 +1,107 @@
+                    if not response.status == 'Success':
+                        raise self.InvalidSessionException
+                    active_token = response["active_token"]
+                    response.session["PANGEA_ACTIVE_TOKEN"] = active_token
+            else:
+                raise self.InvalidSessionException
+        except self.InvalidSessionException as e:
+            request.user = AnonymousUser()
+        else:
+            request.user = self.authn.get_user(username=active_token["email"])
+
+        return self.get_response(request)
+
+class PangeaAuthentication(BaseBackend):
+    def __init__(self):
+        token = os.getenv("PANGEA_AUTHN_TOKEN")
+        domain = os.getenv("PANGEA_DOMAIN")
+        self.config = PangeaConfig(domain=domain)
+        self.authn = AuthN(token, config=self.config, logger_name="pangea")
+        super().__init__()
+
+    def authenticate(self, request):
+        code = request.GET.get('code')
+        user = None
+        if not code:
+            return None
+        resp = self.authn.client.userinfo(code=code)
+        if resp and resp.status == "Success":
+            try:
+                refresh = resp.raw_result['refresh_token']
+                active = resp.raw_result['active_token']
+                user, created = UserModel.objects.get_or_create(username=active["email"])
+                if created:
+                    user.email = active["email"]
+                    user.first_name = active["profile"]["first_name"]
+                    user.last_name = active["profile"]["last_name"]
+                    user.last_login = active["profile"]["Last-Login-Time"]
+                    user.is_active = True
+                    user.save()
+                else:
+                    user.last_login = active["profile"]["Last-Login-Time"]
+                    user.save()
+                request.session["PANGEA_REFRESH_TOKEN"] = refresh
+                request.session["PANGEA_ACTIVE_TOKEN"] = active
+            except Exception as e:
+                print(traceback.format_exc())
+        if not user:
+            request.session["PANGEA_REFRESH_TOKEN"] = None
+            request.session["PANGEA_ACTIVE_TOKEN"] = None
+            request.session["PANGEA_USER"] = None
+        return user
+
+    def logout(self, request):
+        active = request.session.get("PANGEA_ACTIVE_TOKEN")
+        resp = self.authn.client.session.logout(active["token"])
+        del request.session["PANGEA_ACTIVE_TOKEN"]
+        del request.session["PANGEA_REFRESH_TOKEN"]
+
+    def user_can_authenticate(self, user):
+        return getattr(user, "is_active", True)
+
+    def get_user(self, user_id=None, username=None):
+        if not user_id and not username:
+            raise ValueError("user_id or username is required")
+        try:
+            if user_id:
+                user = UserModel._default_manager.get(pk=user_id)
```

Comment:
Why are we using `_default_manager` over `objects` here?

Reply:
i got this from the django auth implementation.